crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -415,6 +430,50 @@ impl BuildSpecCommand {
 			true
 		};
 
+		// Prompt the user for deterministic build only if the profile is Production.
+		let deterministic = deterministic || (profile == Profile::Production && cli
+			.confirm("Would you like to build the runtime deterministically? (Recommended for production builds)")
+			.initial_value(true)
+			.interact()?);
+
+		// If deterministic build is selected, use the provided runtime path or prompt the user if
+		// missing.
+		let runtime_dir = if deterministic {
+			match runtime_dir {
+				Some(dir) => dir,
+				None => PathBuf::from(
+					cli.input("Enter the directory path where the runtime is located:")
+						.placeholder(DEFAULT_RUNTIME_DIR)
+						.default_input(DEFAULT_RUNTIME_DIR)
+						.interact()?,
+				),
+			}
+		} else {
+			DEFAULT_RUNTIME_DIR.into()
+		};
+		// If deterministic build is selected, extract package name from runtime path provided
+		// aboce. Prompt the user if unavailable.
+		let package = if deterministic {
+			match package {
```

Comment:
Can you not simplify with .unwrap_or_else(..) or something similar?

Reply:
Refactored https://github.com/r0gue-io/pop-cli/pull/425/commits/8f7b7973de590997c474a456261c303695093914